core/state_processor.go

```diff
@@ -160,13 +160,12 @@ func applyTransaction(msg *Message, config *params.ChainConfig, gp *GasPool, sta
 // and uses the input parameters for its environment. It returns the receipt
 // for the transaction, gas used and an error if the transaction failed,
 // indicating the block was invalid.
-func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
+func ApplyTransaction(config *params.ChainConfig, bc ChainContext, blockContext vm.BlockContext, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
```

Comment:
switch to passing in `vm.BlockContext` so that we can easily override this for executing a single transaction.

This is used specifically in the worker where we need to populate the `BlockContext` for the execution of a single transaction at a time. We don't want to re-serialize the predicate results and re-construct it within `ApplyTransaction` each time during `NewEVMBlockContext` so this code is modified to take in a `vm.BlockContext` so that we can set the transaction predicate results for each transaction before executing it.